platform/core/src/classes/metadata/StudyMetadata.js

```diff
@@ -303,13 +316,25 @@ export class StudyMetadata extends Metadata {
    * @returns {boolean} Returns true on success or false on failure (e.g., the series does not belong to this study)
    */
   createAndAddDisplaySetsForSeries(sopClassHandlerModules, series) {
-    if (this.containsSeries(series)) {
-      this.setDisplaySets(
-        this._createDisplaySetsForSeries(sopClassHandlerModules, series)
-      );
-      return true;
+    if (!this.containsSeries(series)) {
+      return false;
     }
-    return false;
+
+    const displaySets = this._createDisplaySetsForSeries(sopClassHandlerModules, series)
+
+    this._displaySets.forEach((displaySet, idx) => {
```

Comment:
Possible issue here: deleting current item in forward iteration loop will skip next item from iteration.

Way out could be reverse iteration using `for` loop or `filter`

Reply:
Yeah normally I would use filter but the array has writable: false. Good catch!